src/nat/front_ends/fastapi/job_store.py

```
@@ -83,101 +261,342 @@ def create_job(self,
+            future = client.submit(job_fn, *job_args, key=f"{job_id}-job", **job_kwargs)
+
+            # Store the future in a variable, this allows us to potentially cancel the future later if needed
+            future_var = Variable(name=job_id, client=client)
+            await future_var.set(future)
+
+            if sync_timeout > 0:
+                try:
+                    _ = await future.result(timeout=sync_timeout)
+                    job = await self.get_job(job_id)
+                    assert job is not None, "Job should exist after future result"
+                    return (job_id, job)
```

Ruff (0.12.2)

Consider moving this statement to an `else` block

(TRY300)